plugins/module_utils/identity/keycloak/keycloak.py

```diff
@@ -132,6 +143,79 @@ def get_token(module_params):
     }
 
 
+def isDictEquals(dict1, dict2, exclude=None):
+    """
+    This function compare if tthe first parameter structure, is included in the second.
+    The function use every elements of dict1 and validates they are present in the dict2 structure.
+    The two structure does not need to be equals for that function to return true.
+    Each elements are compared recursively.
+    :param dict1:
+        type:
+            dict for the initial call, can be dict, list, bool, int or str for recursive calls
+        description:
+            reference structure
+    :param dict2:
+        type:
+            dict for the initial call, can be dict, list, bool, int or str for recursive calls
+        description:
+            structure to compare with first parameter.
+    :param exclude:
+        type:
+            list
+        description:
+            Key to exclude from the comparison.
+        default: None
+    :return:
+        type:
+            bool
+        description:
+            Return True if all element of dict 1 are present in dict 2, return false otherwise.
+    """
+    try:
+        if isinstance(dict1, list) and isinstance(dict2, list):
+            if len(dict1) == 0 and len(dict2) == 0:
+                return True
```

Comment:
> So if dict1 is an empty list, but dict2 is not empty, this function will return False? That doesn't really make sense to me.
> 
```suggestion
            if len(dict1) == 0:
                return True
```
I think this would solve the issue ? And doing the same at line 204